src/server/worker/master.go

```diff
@@ -917,7 +917,12 @@ func (a *APIServer) receiveSpout(ctx context.Context, logger *taggedLogger) erro
 						return err
 					}
 					// put files into pachyderm
-					if a.pipelineInfo.Spout.Overwrite || strings.Contains(fileHeader.Name, "marker") {
+					if a.pipelineInfo.Spout.Marker != "" && strings.Contains(fileHeader.Name, a.pipelineInfo.Spout.Marker) {
```

Comment:
should this be a stricter check than containment (e.g. equality)? what if we emit a file that happens to have the marker filename as a substring?

Reply:
That's a good point, we'll make the check here a bit narrower.